### PR TITLE
Improvement FEMR53 MissionTripFilter

### PR DESCRIPTION
**Pull request by akrebs1202**
The MissionTripFilter dropdown in the Research is now complete. The dropdown populates the mission trips, displays the appropriate information when selected, and upon clicking the clear button it will reset the users selection.

### CODE REVIEW

File: app/femr/common/dtos/ServiceResponse.java [hunk 1]
```diff
@@ -18,10 +18,12 @@
 */
 package femr.common.dtos;
 
+import femr.common.models.MissionTripItem;
+
 import java.util.HashMap;
 import java.util.Map;
 
-public class ServiceResponse<T> {
+public class ServiceResponse<T> extends MissionTripItem {
```
Comment (kdunlap) [inline app/femr/common/dtos/ServiceResponse.java]: I'm not entirely sure how this got here. It definitely is not necessary, it should be:

`public class ServiceResponse<T> {`

File: app/femr/data/models/mysql/PatientEncounter.java [hunk 2]
```diff
@@ -56,7 +56,7 @@
     @JoinColumn(name = "patient_age_classification_id")
     private PatientAgeClassification patientAgeClassification;
     @ManyToOne(fetch = FetchType.EAGER)
-    @JoinColumn(name = "mission_trip_id")
+    @JoinColumn(name = "mission_city_id")
```
Comment (kdunlap) [inline app/femr/data/models/mysql/PatientEncounter.java]: The patient encounter column for trip is mission_trip_id, this should be reverted back

`@JoinColumn(name = "mission_trip_id")`

**Comment by kdunlap** [comment]
@akrebs1202: just 2 quick things to fix up on this PR and it looks good to merge otherwise

**Comment by kdunlap** [comment]
The only things I can think of, which may warrant an additional task:

1. The Mission Trip dropdown on research should also have the date

    - It currently shows: Team Name - Country - City
    - The same team will go to the same country/city mutliple different times. On my install I show  World     Health Student Organization - Haiti - Morne De L' Hopital 3 separate times.


2. Should we be able to select multiple trips at a time?

**Comment by kdunlap** [comment]
Manually squashed, rebased and merged

https://teamfemr.atlassian.net/browse/FEMR-53

**Comment by akrebs1202** [comment]
Ken,
The date could be added in the filter for ease of use. For testing purposes I used the filter date and that worked as well. I noticed you closed it, is this additional task something that you were planning on implementing next semester or would you like me to fix it and push it? 


    On Saturday, December 12, 2015 1:52 PM, Ken Dunlap <notifications@github.com> wrote:
 

 Closed #465.—

**Comment by kdunlap** [comment]
I merged your code from this pull request. I need to make a new issue to add the dates and such. If you'd like to implement it - that would be totally awesome :thumbsup: It would need to be a separate pull request though

**Pull request closed without merging**